scripts/cvmfsexec-osg-wrapper

```diff
@@ -114,30 +105,39 @@ if [[ -n $CVMFS_HTTP_PROXY ]]; then
     add_or_replace "$cvmfs_local_config" CVMFS_HTTP_PROXY "${CVMFS_HTTP_PROXY}"
 fi
 
-"$workdir"/cvmfsexec/umountrepo -a || :
+# If we have to use this script to mount CVMFS, we can't use images from there anyway.
+export ALLOW_NONCVMFS_IMAGES=true
 
-trap '"$workdir"/cvmfsexec/umountrepo -a; cd "$prevdir"; rm -rf "$workdir"' EXIT
+echo >&2 "*** Running cvmfsexec smoke test:"
+if "$workdir"/cvmfsexec/cvmfsexec -N -- /bin/true; then
+    echo >&2 "*** cvmfsexec smoke test passed: you have the permissions to run cvmfsexec directly"
 
-# Mount repos
-for repo in config-osg.opensciencegrid.org $CVMFSEXEC_REPOS; do
-    "$workdir"/cvmfsexec/mountrepo "$repo" || fail "Unable to mount cvmfs repo $repo"
-done
+    # Note: do not use 'exec'; cleanup won't run
+    export CVMFS_BASE=/cvmfs
+    export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    echo "*** cvmfs available at $CVMFS_BASE" >&2
+    "$workdir"/cvmfsexec/cvmfsexec -N $CVMFSEXEC_REPOS -- "$@"
+else
+    echo >&2 "*** cvmfsexec smoke test failed: will use mountrepo instead"
+    command -v fusermount >& /dev/null || fail "Required command 'fusermount' not found"
 
-export CVMFS_BASE="$workdir"/cvmfsexec/dist/cvmfs
+    "$workdir"/cvmfsexec/umountrepo -a || :
```

Comment:
I'm a little lost, what's this for exactly?

Reply:
In case there are leftover mounts from a previous run that never had time to unmount.

Thinking about it, we probably shouldn't do this unless our jobs are whole-machine jobs.

Reply:
At the very least, could you add a comment?